alert/tests.py

```diff
@@ -532,21 +534,341 @@ def test_no_permission(self):
+        self.assertEqual(response.data['profile']['email'], None)
+        self.assertEqual(response.data['profile']['phone'], '+13131234567')
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_null_phone(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob',
+                                               'profile': {'email': 'example@email.com', 'phone': None}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], None)
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], 'example@email.com')
+        self.assertEqual(response.data['profile']['phone'], None)
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_both_null(self):
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob',
+                                               'profile': {'email': None, 'phone': None}}),
+                                   content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['profile']['email'], None)
+        self.assertEqual(response.data['profile']['phone'], None)
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+        response = self.client.get('/api/settings/')
+        self.assertEqual(200, response.status_code)
+        self.assertEqual(response.data['profile']['email'], None)
+        self.assertEqual(response.data['profile']['phone'], None)
+        self.assertEqual(response.data['username'], 'jacob')
+        self.assertEqual(response.data['first_name'], '')
+        self.assertEqual(response.data['last_name'], '')
+
+    def test_multiple_users_independent(self):
+        User.objects.create_user(username='murey',
+                                 password='top_secret')
+        client2 = APIClient()
+        client2.login(username='murey', password='top_secret')
+        response = self.client.put('/api/settings/',
+                                   json.dumps({'username': 'jacob',
```

Comment:
do you need to always include `username` in the `PUT` requests to an account?

Reply:
Fixed that.